### DISCUSSION TITLE
2.4.60 Distributed on ESXI - Forward Sensors in Separate Subnet Failing to Start

### DISCUSSION THREAD

**Discussion by hoschmidt2020** [Security-Onion-Solutions/securityonion#13557] [2.4]
### Version

2.4.60

### Installation Method

Security Onion ISO image

### Description

other (please provide detail below)

### Installation Type

Distributed

### Location

airgap

### Hardware Specs

Exceeds minimum requirements

### CPU

10

### RAM

32

### Storage for /

350

### Storage for /nsm

82

### Network Traffic Collection

tap

### Network Traffic Speeds

Less than 1Gbps

### Status

Yes, all services on all nodes are running OK

### Salt Status

Yes, there are salt failures (please provide detail below)

### Logs

Yes, there are additional clues in /opt/so/log/ (please provide detail below)

### Detail

I have been attempting to establish a distributed deployment of Security Onion 2.4.60 on an ESXI server. I have the following setup for my networks and nodes:

192.168.2.0/24 - Management Subnet
192.168.2.50 - Manager Node
192.168.2.51 - Search Node
192.168.2.52 - Sensor Node

192.168.1.0/24 - IT Subnet
192.168.1.50 - Sensor Node

192.168.11.0/29 - DNS DMZ
192.168.11.4 - Sensor Node

192.168.11.8/29 - Web DMZ
192.168.11.12 - Sensor Node

192.168.11.16/29 - Web DMZ
192.168.11.19 - Sensor Node

Each of these subnets have a pfSense firewall in-between them, however there is currently an ANY rule in place to allow all traffic for the time being until everything is configured. Each of the virtual subnets has proper routing between each of them, meaning that you are able to ping any machine from any other subnet. 

I have been able to install and configure all of the sensors, and accept them into the grid via the Security Onion Console, however I have run into an issue now where the sensors outside of the immediate subnet are failing to start correctly. Before the IT subnet sensor was able to connect properly, and I could see it's status on the grid page, but after a reboot services started to fail. I have since removed each of the outward sensors (not in the managers subnet) from the grid and reinstalled each of them individually. However, the same errors persists.

One of a couple things will happen when these outward sensors restart. Either it seems to be stuck in a starting phase:
![image](https://github.com/user-attachments/assets/771d7888-894a-4da8-beb0-dba0228e5a5c)
Where a `sudo salt-call state.highstate` provides:
![image](https://github.com/user-attachments/assets/8277ac33-ab83-47c4-9245-35645298b0f9)

Or it will simply state that `so-status is not yet available` with the same salt-call output as previously shown. Or in another case:
![image](https://github.com/user-attachments/assets/2f669670-08f2-405f-94c6-f9b6d174b890)

Running `sudo docker ps` in many cases none of the containers are running, or in one instance strelka-backend was failing to start, which coincided with the following console messages:
![image](https://github.com/user-attachments/assets/8715b0ca-9523-45c6-b572-eaefc89352fb)
 
Going through the logs many are empty, the little I've seen show errors for missing files for Strelka: 
![image](https://github.com/user-attachments/assets/944df7cd-5d73-4ebb-883e-bf82ce962c60)

I am wondering if this is a networking issue? As a test I created a fresh sensor in the immediate subnet and after the configuration and being accepted into the grid it started immediately without issue. I've seen some comments in other discussions on ESXI about not having enough resources but after expanding the manager nodes resources and others I haven't seen a difference. The virtual switch and port groups on ESXI are setup with the correct settings used by others and noted in the SO documentation.

Another likely innocuous thing I've been noticing is the presence of these dropped multicast packets by the SO manager node:
![image](https://github.com/user-attachments/assets/4b6c5f14-25cc-4cc9-8190-96bd7cea6998)

I'm looking to get more of an insight into how the Security Onion sensors connect to the manager nodes in order to fix these failing sensors. They obviously have the ability to connect since the configuration and grid acceptance worked, as well as pings between the two, but something else is failing.

### Guidelines

- [X] I have read the discussion guidelines at https://github.com/Security-Onion-Solutions/securityonion/discussions/1720 and assert that I have followed the guidelines.

**Comment by hoschmidt2020** [comment]
Small update after being left running for awhile, two of the forward sensors popped up with this message in the console:
![image](https://github.com/user-attachments/assets/271347b7-e001-4d34-b144-a3e307d2b44c)

While another when I called `sudo salt-call state.highstate` changed to the following output:
![image](https://github.com/user-attachments/assets/94127b3c-af09-4dff-aeb1-548e7a4c873a)

**Reply by hoschmidt2020** [reply]
Left the machines running overnight, it looks like somehow the installation is failing to install everything correctly? On three of the sensors the docker daemon isn't even running, on the forward-IT sensor which said installation wasn't complete only two containers are up while the rest are missing.
![image](https://github.com/user-attachments/assets/435e3227-0319-4a2e-ab3e-5207644c3247)

**Comment by reyesj2** [comment]
Have you verified this sensor can access the manager on the ports listed at https://docs.securityonion.net/en/2.4/firewall.html#node-communication 
From your sensor you could run ```nc -zv <manager_ip> <port>```
